app/views/schedule/_program_session.html.haml

```diff
@@ -1,11 +1,23 @@
 .schedule-grid-cell
-  - if !program_session.display_track_name.blank?
-    .session-track
-      = program_session.display_track_name
-  .schedule-session-title
-    = program_session.display_title
-  .schedule-room
-    %span.room-icon
-    = program_session.room
-  .schedule-presenter
-    =  program_session.display_presenter
+  .schedule-title-room-wrapper
+    %a.schedule-session-title{ href: "#", data: { action: 'click->fly-out#updateSidePanel', 'session-id': dom_id(program_session) } }
+      = program_session.display_title
+    .schedule-room
+      = image_tag 'themes/default/room-icon.png', class: 'room-icon'
+      = program_session.room
+  .schedule-speaker-tag-wrapper
+    .schedule-presenter
+      =  program_session.display_presenter
+
+    .schedule-tags
+      - if program_session.sponsored?
+        %span.sponsored-tag Sponsored
+
+      - if program_session.filled_with_session?
+        %span.session-format-tag{ class: "#{current_website.session_format_tag_class(program_session.session_format)}" }
+          = current_website.session_format_name(program_session.session_format)
+
+      -if !program_session.display_track_name.blank?
```

Comment:
would `program_session.display_track_name.present?` work here?

Reply:
Why yes it would. ANNND this also one of those early poor naming decisions that I fixed last night. This is a the object being iterated over is a `time_slot`, which makes it more clear why the decorator method is required.